indra/newview/lldrawpool.cpp

```diff
@@ -688,6 +671,81 @@ bool LLRenderPass::uploadMatrixPalette(LLVOAvatar* avatar, LLMeshSkinInfo* skinI
     return true;
 }
 
+// Returns true if rendering should proceed
+//static
+bool LLRenderPass::uploadMatrixPalette(LLVOAvatar* avatar, LLMeshSkinInfo* skinInfo, const LLVOAvatar*& lastAvatar, U64& lastMeshId, bool& skipLastSkin)
```

Comment:
Inspection of `LLDrawPoolMaterials` indicated that there are several important things to check when drawing a rigged mesh:

- Is the avatar null? If so, don't render.
    - Note: Some cases of null avatar are more suspicious than others. Some may be bugs. I previously added `llassert(avatar)` here for a bit to catch some bugs, but I'm not convinced all cases of null avatar are bugs.
- Is the matrix palette not defined? If so, don't render.
- During alpha draw passes, shaders are interleaved, so don't assume the skinning is up-to-date if the shader changes